File: game_bizModule/src/com/game/bizModule/login/msg/CGLogin.java
package com.game.bizModule.login.msg;

import com.game.bizModule.global.AllMsgSerialUId;
import com.game.bizModule.login.handler.Handler_CGLogin;
import com.game.gameServer.msg.AbstractCGMsgObj;
import com.game.part.msg.type.MsgStr;

/**
 * 登陆游戏
 * 
 * @author hjj2019
 *
 */
public class CGLogin extends AbstractCGMsgObj<Handler_CGLogin> {
	/** 登陆字符串 */
	public MsgStr _loginStr;

	/**
	 * 类默认构造器
	 *
	 */
	public CGLogin() {
	}

	/**
	 * 类参数构造器
	 *
	 * @param loginStr
	 *
	 */
	public CGLogin(String loginStr) {
		this._loginStr = new MsgStr(loginStr);
	}

	@Override
	public short getSerialUId() {
		return AllMsgSerialUId.CG_LOGIN;
	}

	@Override
	public Handler_CGLogin getSelfHandler() {
		return new Handler_CGLogin();
	}
}

File: client_wpf/Xgame.GameBizModule/Login/Msg/CGLogin.cs
using System;

using Xgame.GameBizModule.Global;
using Xgame.GameClient.Msg;
using Xgame.GamePart.Msg.Type;

namespace Xgame.GameBizModule.Login.Msg
{
    /// <summary>
    /// 登陆 CG 消息
    /// </summary>
    public partial class CGLogin : BaseCGMsg
    {
        /** 平台 UId */
        public MsgStr _platformUId;
        /** 登陆字符串, JSON 格式 */
        public MsgStr _loginStr;

        // @Override
        public override short MsgSerialUId
        {
            get
            {
                return AllMsgSerialUId.CG_LOGIN;
            }
        }
    }
}
